src/SoundClassifier/index.js

```diff
@@ -0,0 +1,90 @@
+// https://opensource.org/licenses/MIT
+
+/*
+Sound Classifier using pre-trained networks
+*/
+
+import * as tf from '@tensorflow/tfjs';
+import handleArguments from "../utils/handleArguments";
+import * as speechCommands from './speechcommands';
+import callCallback from '../utils/callcallback';
+
+const MODEL_OPTIONS = ['speechcommands18w'];
+class SoundClassifier {
+    /**
+     * Create an SoundClassifier.
+     * @param {modelNameOrUrl} modelNameOrUrl - The name or the URL of the model to use. Current name options
+     *    are: 'SpeechCommands18w'.
+     * @param {object} options - An object with options.
+     * @param {function} callback - A callback to be called when the model is ready.
+     */
+    constructor(modelNameOrUrl, options, callback) {
+        this.model = null;
+        this.options = options;
+        if (typeof modelNameOrUrl === 'string') {
+            if (MODEL_OPTIONS.includes(modelNameOrUrl)) {
+                this.modelName = modelNameOrUrl;
+                this.modelUrl = null;
+                switch (this.modelName) {
+                    case 'speechcommands18w':
+                        this.modelToUse = speechCommands;
+                        break;
+                    default:
+                        this.modelToUse = null;
+                }
+            } else {
+                // Default to speechCommands for now
+                this.modelToUse = speechCommands;
+                this.modelUrl = modelNameOrUrl;
+            }
+        }
+        // Load the model
+        this.ready = callCallback(this.loadModel(options, this.modelUrl), callback);
+    }
+
+    async loadModel(options) {
+        this.model = await this.modelToUse.load(options, this.modelUrl);
+        return this;
+    }
+
+    async classifyInternal(numberOfClasses, callback) {
+        // Wait for the model to be ready
+        await this.ready;
+        await tf.nextFrame();
+
+        return this.model.classify(numberOfClasses, callback);
+    }
+
+    /**
+     * Classifies the audio from microphone and takes a callback to handle the results
+     * @param {function | number} numOrCallback -
+     *    takes any of the following params
+     * @param {function} cb - a callback function that handles the results of the function.
+     * @return {function} a promise or the results of a given callback, cb.
+     */
+    async classify(numOrCallback = null, cb) {
+        const args = handleArguments(numOrCallback, cb);
+        const numberOfClasses = args.number || this.topk;
+
+        return this.classifyInternal(numberOfClasses, args.callback);
+    }
+}
+
+const soundClassifier = (modelName, optionsOrCallback, cb) => {
+    const { string, options = {}, callback } = handleArguments(modelName, optionsOrCallback, cb)
+        .require('string', 'Please specify a model to use. E.g: "SpeechCommands18w"');
+
+    let model = string;
+    // TODO: I think we should delete this. -Linda
```

Comment:
Since this model is getting some testing presently, now is probably a good time to follow Linda's suggestion here, and see if anything breaks!

Reply:
I tried running with it deleted and got an error message "Unsupported URL scheme in metadata URL:..."

These convert 'model' from **'SpeechCommands18w' to 'speechcommands18w'**
So I tried deleting the lines again but also changing the initializer line in the example code from **uppercase to lowercase** 'classifier = await ml5.soundClassifier("**speechcommands18w**", options);' This works.

I wonder what she meant by delete. Perhaps, she means to consider whether there is a better solution?

Reply:
Thank you for figuring out what this was for, @ch3926!

And yes, I think we should keep it: the library assumes all lower-case, but existing code [seems to use](https://learn.ml5js.org/#/reference/sound-classifier?id=quickstart) camel-case. Perhaps better solving this in the constructor, though, with a helpful comment.

Reply:
@ch3926 I think I was interpreting the `if` statement backwards when I left that comment! 🙃 We do not want to modify the casing if the user provides a URL.  So you can delete my comment.